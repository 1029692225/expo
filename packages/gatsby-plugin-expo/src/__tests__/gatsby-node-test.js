const {
  resolvableExtensions,
  onCreateBabelConfig,
  onCreateWebpackConfig,
} = require(`../gatsby-node`);

describe(`resolvableExtensions`, () => {
  it(`returns the correct resolvable extensions`, () => {
    expect(resolvableExtensions()).toMatchSnapshot();
  });
});

describe(`onCreateBabelConfig`, () => {
  it(`sets the correct babel preset`, () => {
    const actions = { setBabelPreset: jest.fn() };
    const options = {
      // TODO: Bacon: try to pass in native props too
    };
    onCreateBabelConfig({ actions }, options);
    expect(actions.setBabelPreset).toHaveBeenCalledWith({
      name: expect.stringContaining('babel-preset-expo'),
      options,
    });
  });
});

describe(`onCreateWebpackConfig`, () => {
  it(`invokes the correct webpack config props`, () => {
    const actions = { setWebpackConfig: jest.fn() };
    const jsLoader = {};
    const getConfig = jest.fn();
    onCreateWebpackConfig({ actions, getConfig });
    expect(actions.setWebpackConfig).toHaveBeenCalled();
    expect(getConfig).toHaveBeenCalled();
  });
});
