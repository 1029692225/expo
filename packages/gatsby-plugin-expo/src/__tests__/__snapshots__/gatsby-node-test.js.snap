// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolvableExtensions returns the correct resolvable extensions 1`] = `
Array [
  ".web.ts",
  ".ts",
  ".web.tsx",
  ".tsx",
  ".web.js",
  ".js",
  ".web.jsx",
  ".jsx",
  ".json",
]
`;
